accounts/modelbackend.py

```diff
@@ -23,11 +23,28 @@
 
 
 class CustomModelBackend(ModelBackend):
-    def authenticate(self, username=None, password=None):
+
+    def authenticate(self, request, username=None, password=None, **kwargs):
         """ authenticate against case insensitive username """
+
+        if '@' in username:
+            # In this case, user is most probably using email+password pair to login
+            # Try to get user object from password and check password
+            try:
+                user = User.objects.get(email__iexact=username)
```

Comment:
That I wanted to comment with you. I checked that and I think we do, but I'm not 100% sure.